component/prometheus/scrape/scrape_test.go

```diff
@@ -9,12 +9,71 @@ import (
 	"github.com/grafana/agent/component"
 	"github.com/grafana/agent/component/prometheus"
 	"github.com/grafana/agent/pkg/flow/logging"
+	"github.com/grafana/agent/pkg/river"
 	prometheus_client "github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/prometheus/model/labels"
 	"github.com/prometheus/prometheus/storage"
 	"github.com/stretchr/testify/require"
 )
 
+func TestRiverConfig(t *testing.T) {
+	var exampleRiverConfig = `
+	targets         = [{ "target1" = "target1" }]
+	forward_to      = []
+	scrape_interval = "10s"
+	job_name        = "local-flow"
+
+	bearer_token = "token"
+	bearer_token_file = "/path/to/file.token"
+	proxy_url = "http://0.0.0.0:11111"
+	follow_redirects = true
+	enable_http2 = true
+
+	basic_auth {
```

Comment:
I feel like it is good practice to test the blown up river config at least happy path in the tests for all these. With that, do we want to put a happy path http_client_config river format somewhere rather than copy paste in every test file. Same thought for tls_config and maybe the others.

Reply:
Yeah, I'm not sure honestly. Testing shared config in a common place might not be necessary. 

I don't disagree that we should have some kind of unmarshaling test, but I'm not sure of the value in testing every possible field, especially for components like these where specific blocks and arguments are mutually exclusive.